src/latexify/codegen/function_codegen_match_test.py

```diff
@@ -184,3 +184,112 @@ def test_matchvalue_mutliple_statements() -> None:
         match=r"^Match cases must contain exactly 1 return statement\.$",
     ):
         function_codegen.FunctionCodegen().visit(tree)
+
+
+@test_utils.require_at_least(10)
+def test_visit_match_case_with_if() -> None:
+    tree = ast.parse(
+        textwrap.dedent(
+            """
+            match x:
+                case x if x > 0:
+                    return 1
+                case _:
+                    return 2
+            """
+        )
+    ).body[0]
+
+    assert (
+        function_codegen.FunctionCodegen().visit(tree)
+        == r"\left\{ \begin{array}{ll}"
+        + r"1, & \mathrm{if} \ x > 0 \\ "
+        + r"2, & \mathrm{otherwise} \end{array} \right."
```

Comment:
As commented, this output is weird, and we shouldn't support guards at this point.